### PR TITLE
CMR-7841 fixing inability to index very long fields in elasticsearch for keyword field

### PR DESCRIPTION
**Pull request by jaybarra**
Excessively long values in elasticsearch exceeded the maximum byte length supported by the text field type and could not be indexed. 
* Update how keywords are generated by changing the parsing of groups
* Split very long terms apart
* Truncate very long text values and warn when doing so

### CODE REVIEW

File: indexer-app/src/cmr/indexer/data/concepts/collection/keyword.clj [hunk 1]
```diff
@@ -67,7 +78,12 @@
...
-         (map #(str " " % " "))
+         (map #(str (when-not (str/starts-with? " " %) " ")
+                    %
+                    (when-not (str/ends-with? " " %) " ")))
          ;; The keyword-in-words here are used for unquoted keyword search. It's exactly the same as the existing
          ;; keyword index fields, without the need to go through a whitespace analyzer.
-         (concat keywords-in-words))))
+         (concat keywords-in-words)
+         (remove str/blank?)
+         (map #(split-large-keywords % 512)) ;; break apart very long values
```
Comment (ygliuvt) [inline indexer-app/src/cmr/indexer/data/concepts/collection/keyword.clj:88]: def `512` as a var.

**Review by ygliuvt — APPROVED** [review]

File: indexer-app/src/cmr/indexer/data/concepts/collection/keyword.clj [hunk 2]
```diff
@@ -3,11 +3,25 @@
   (:require
    [clojure.string :as str]
```
Comment (eereiter) [inline indexer-app/src/cmr/indexer/data/concepts/collection/keyword.clj:4]: This should be changed from str to string.  I noticed that you are using str below in your code.

File: indexer-app/test/cmr/indexer/test/data/concepts/keyword_util.clj [hunk 3]
```diff
@@ -1,10 +1,12 @@
 (ns cmr.indexer.test.data.concepts.keyword-util
   "Functions for testing cmr.indexer.data.concepts.keyword-util namespace."
   (:require
+   [clojure.string :as str]
```
Comment (eereiter) [inline indexer-app/test/cmr/indexer/test/data/concepts/keyword_util.clj:4]: change str to string

**Review by eereiter — APPROVED** [review]
Please change the two file str to string.

**Pull request merged**